app/src/main/kotlin/com/example/app/agents/ElvenAgent.kt

```
@@ -82,93 +81,120 @@ class ElvenAgent(
+                logger.trace("Running command: {}", input)
+
+                agent.run(input)
+                agent.close()
+
+                awaitClose {
+                    logger.debug("Flow cancelled for session: $chatSessionId, cleaning up resources")
+                    runBlocking {
+                        try {
+                            agent.close()
+                            logger.debug("Agent closed successfully for session: $chatSessionId")
+                        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)